### PR TITLE
Custom label feature added 

### LINKED ISSUE #28: Provide an API to enable users to pass custom labels
## Allow users to pass custom labels from the API

```[tasklist]
### CustomRulerLabel
- [ ] https://github.com/GeekyAnts/GaugesFlutter/issues/40
- [ ] https://github.com/GeekyAnts/GaugesFlutter/issues/41
```

### API
```dart
        customRulerLabel : <CustomRulerLabel>[
            const CustomRulerLabel(text: '$0', value: 0),
            const CustomRulerLabel(text: '$10', value: 100),
            const CustomRulerLabel(text: '$20', value: 200),
            const CustomRulerLabel(text: '$25', value: 250),
            const CustomRulerLabel(text: '$30', value: 300),
          ];
        },
```

which could result in something like this 

Screenshot 2023-03-03 at 12 01 25 PM

### PR DESCRIPTION
**Pull request by SahilSharma2710**
This PR closes or fixes #28 issue
Now user can add CustomLabels to the linear gauge (shown in image below)
Screenshot 2023-03-05 at 10 54 33 PM

When users added custom labels starting and ending value of the linear gauge is set up according to the user custom labels 
Screenshot 2023-03-05 at 10 47 46 PM

For this addCustomLabels function is added in linear_gauge_label.dart file
Screenshot 2023-03-05 at 10 49 32 PM

To create offset for labels  new mathematical expression is created
Screenshot 2023-03-05 at 10 50 22 PM

### CODE REVIEW

File: lib/linear_gauge/custom_label/custom_linear_gauge_label.dart [hunk 1]
```diff
@@ -0,0 +1,40 @@
+class CustomLinearGaugeLabel {
```
Comment (hasnentai) [inline lib/linear_gauge/custom_label/custom_linear_gauge_label.dart:1]: I think we need a small change here 
can we change the class name to `RulerCustomLabel`? I think that make more sense .
What do you think ?

**Review by hasnentai — CHANGES_REQUESTED** [review]

File: lib/linear_gauge/custom_label/custom_linear_gauge_label.dart [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  /// A [CustomLinearGaugeLabel] is a widget that allows to add custom labels and values on the [LinearGauge].
+  ///
+  /// ```dart
+  /// child : const LinearGauge(
+  /// customLabels: [
+  /// CustomLinearGaugeLabel(text: "1%", value: 10),
```
[the remark this answers is not in the document]
Comment (hasnentai) [reply lib/linear_gauge/custom_label/custom_linear_gauge_label.dart:7]: Once we change the class name we need to change every where in the example and files

File: lib/linear_gauge/custom_label/custom_linear_gauge_label.dart (hunk 1 above)
Comment (SahilSharma2710) [reply lib/linear_gauge/custom_label/custom_linear_gauge_label.dart:1]: Yes agreed that will make more sense . Custom labels are for all gauges not only for linear. Will change that.

File: lib/linear_gauge/custom_label/custom_linear_gauge_label.dart (hunk 2 above)
[the remark this answers is not in the document]
Comment (hasnentai) [reply lib/linear_gauge/custom_label/custom_linear_gauge_label.dart:7]: Or how about maybe `CustomRulerLabel`? Lets try to keep RulerLabel together

File: lib/linear_gauge/custom_label/custom_linear_gauge_label.dart (hunk 1 above)
Comment (hasnentai) [reply lib/linear_gauge/custom_label/custom_linear_gauge_label.dart:1]: Or how about maybe CustomRulerLabel? Lets try to keep RulerLabel together

Comment (SahilSharma2710) [reply lib/linear_gauge/custom_label/custom_linear_gauge_label.dart:1]: As per terminology we used , labels are part of rulers only. So `CustomRulerBar` or `CustomRuler` can also work but for user `CustomRulerLabel` make more sence . So what should be the final call

**Review by hasnentai — APPROVED** [review]
202313172-687ccda4-96b5-47c4-b8e5-fe653dd442c7

**Pull request merged**